Expose git_diff_blobs API

staktrace opened the pull request:
Adds a `diff_blobs` method on the `Repository` which exposes the `git_diff_blobs` API in libgit2. The patch should be functionally complete although I haven't tested it extensively. The example app changes could be improved, but I wanted to get some feedback first as to whether this patch would be acceptable or not, and if any more fundamental changes are needed.

Thanks!

staktrace commented:
Hm, looks like I wrote my patches against alex's fork which is stale. I can rebase against the master, but would like feedback on the main approach first.

staktrace commented:
Patches rebased to master.

alexcrichton commented:
Seems reasonable to me! Usage-wise a builder might be a bit more ergonomic than a many-argument-method, but I don't really have much of a preference one way or another.

staktrace commented:
Patches ready for review now. I ran rustfmt and modified the `diff` example a bit more to allow colorized output for the `--blobs` option that I added.
